iceoryx_hoofs/README.md

```diff
@@ -39,112 +22,136 @@ and we do not use dynamic memory. In these cases we adjusted the API to our use
 Most of the headers are providing some example code on how the
 class should be used.
 
-| class/file            | internal | description                                                                                                                                                                                                                           |
+The module structure is a logical grouping. It is replicated for `concurrent` and `posix` implementations.
+
+### Memory & lifetime management (memory)
+
+| class                              | internal | description                                                                                                                                                                                                              |
+|:----------------------------------:|:--------:|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|`unique_ptr`                        |          | Provides a heap-less unique_ptr implementation, unlike the STL                                                                                                                                                           |
+|`RelativePointer`                   |          | Pointer which can be stored in shared memory                                                                                                                                                                             |
+|`ScopeGuard`                        |          | This is an abstraction of the C++ RAII idiom. Sometimes you have constructs where you would like to perform a certain task on creation and then again when they are getting out of scope, this is where `ScopeGuard` comes in. It is like a `std::lock_guard` or a `std::shared_ptr` but more generic. |
+|`scoped_static`                     |          | Helper function to limit lifetime of static or global variables to a scope                                                                                                                                               |
+|`static_storage`                    | i        | Untyped aligned static storage.                                                                                                                                                                                          |
+|`shared_memory_object/Allocator`    | i        | Helper class for the `SharedMemoryObject`.                                                                                                                                                                               |
```

Comment:
At some point this should be moved out of shared_memory_object and renamed to bump_allocator